swcc/swcc/models/dataset.py

```diff
@@ -0,0 +1,261 @@
+from __future__ import annotations
+
+from pathlib import Path
+import re
+
+try:
+    from typing import Dict, Iterator, Literal, Optional, Union
+except ImportError:
+    from typing import (
+        Dict,
+        Iterator,
+        Optional,
+        Union,
+    )
+    from typing_extensions import (  # type: ignore
+        Literal,
+    )
+
+from openpyxl import load_workbook
+from pydantic import BaseModel
+
+from .api_model import ApiModel
+from .file_type import FileType
+from .utils import FileIO, NonEmptyString, logger, shape_file_type
+
+
+class DataFileIO(BaseModel, FileIO):
+    dataset: Dataset
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    def load_data(self, interpret=True):
+        if (
+            not self.dataset.file
+            or not hasattr(self.dataset.file, 'path')
+            or not self.dataset.file.path
+        ):
+            return
+        file = self.dataset.file.path
+        data = None
+        if str(file).endswith('xlsx'):
```

Comment:
Generally, excel files can also have .xls.  I don't know if we also can handle csv files.  Do we want to include those as well?

Reply:
`openpyxl` raises an exception when passed a csv, saying that it's not supported, so we'll continue to not support csv. [bc9f4d2](https://github.com/girder/shapeworks-cloud/pull/185/commits/bc9f4d2b1800c835c4a3dc9b165199c5d52ddc9d) includes accepting `xls`

Reply:
I found that [bc9f4d2](https://github.com/girder/shapeworks-cloud/pull/185/commits/bc9f4d2b1800c835c4a3dc9b165199c5d52ddc9d) failed its tests because `csv` files were passed in and rejected. I tried [3e9a6d4](https://github.com/girder/shapeworks-cloud/pull/185/commits/3e9a6d4ef0041132bc9478ea37a604e995cc4f77), but the expected `openpyxl` error occurs. What were we doing previously to accept csv files?

Reply:
Maybe we aren't doing anything with the fake csv files...  Do the tests pass if you just change the faker call to use xlsx instead of csv?

Reply:
It doesn't look like Faker can correctly create these `xlsx` files. Am I doing it wrong, or should we instead point the factory to a few example files?

Reply:
This PR is changing a behavior of the Project model.  Before the `Project.create` method wasn't explicitly specified, and therefore the file wasn't parsed on create.  Instead, it is parsed during `Dataset.add_project`.  In the new code, it is called during create, so the tests that check the Project model fail with a csv.  Ultimately, this means that the test with the ProjectFactory was never actually using the created csv, though it is requiring that the file field be populated.

Reply:
Interesting. However, it seems that having the Project file parsed during the `create` method is probably desired behavior, and not doing so isn't testing the parsing of the project file. [0ff1eb5](https://github.com/girder/shapeworks-cloud/pull/185/commits/0ff1eb552a9a53bbf75b5013235456a43877cdf5) uses an example project file in the Project Factory that will be correctly parsed.

Reply:
I concur that parsing during creation is more correct.